### PR TITLE
CASL support

### PR DESCRIPTION
**Pull request by cedporter**
This is an initial attempt at CASL support, a [scripting language](https://go.documentation.sas.com/?cdcId=pgmsascdc&cdcVersion=9.4_3.4&docsetId=pgmdiff&docsetTarget=p06ibhzb2bklaon1a86ili3wpil9.htm&locale=en) used within PROC CAS in SAS.

### CODE REVIEW

**Comment by cedporter** [comment]
Any update on a review of this PR?

File: components/prism-sas.js [hunk 1]
```diff
@@ -22,19 +22,31 @@
 		}
 	];
 
+	var mVariable = {
```
Comment (RunDevelopment) [inline components/prism-sas.js]: 1. Generic name.
2. It's only used once, so why not inline it?

File: components/prism-sas.js [hunk 2]
```diff
@@ -57,6 +92,34 @@
...
+
+	var casActions = {
+		pattern: RegExp('(^|\\s+)(?:action(?:\\s)+)?(?:<act>)(?:\\.[a-z]+\\b)(?:[\\s\\S]+?)(?=;)'.replace(/<act>/g, actionSets), 'i'),
```
Comment (RunDevelopment) [inline components/prism-sas.js]: 1. For a lookbehind `(?<=\s+)` == `(?<=\s)`
2. A few unnecessary non-capturing groups.
3. `(?:[\\s\\S]+?)(?=;)` == `[^;]+`

```suggestion
		pattern: RegExp('(^|\\s)(?:action\\s+)?(?:<act>)\\.[a-z]+\\b[^;]+'.replace(/<act>/g, actionSets), 'i'),
```

File: components/prism-sas.js [hunk 3]
```diff
@@ -57,6 +92,34 @@
...
+		lookbehind: true,
+		inside: {
+			'keyword': RegExp('(?:<act>)(?:\\.[a-z]+\\b)'.replace(/<act>/g, actionSets), 'i'),
```
Comment (RunDevelopment) [inline components/prism-sas.js]:
```suggestion
			'keyword': RegExp('(?:<act>)\\.[a-z]+\\b'.replace(/<act>/g, actionSets), 'i'),
```

File: components/prism-sas.js [hunk 4]
```diff
@@ -130,6 +193,43 @@
 			}
 		},
 
+		'proc-cas': {
+			pattern: /(^proc\s+cas(?:\s+[\w|=]+)?;)[\s\S]+?(?=^(?:proc\s+\w+|quit|data);|(?![\s\S]))/im,
+			lookbehind: true,
+			inside: {
+				'statement-var': {
+					pattern: /((?:^|\s)=?)(?:saveresult)(?:\s)+(?:[\s\S]+?)(?=;)/im,
```
Comment (RunDevelopment) [inline components/prism-sas.js]:
```suggestion
					pattern: /((?:^|\s)=?)saveresult\s+[^;]+/im,
```

File: components/prism-sas.js [hunk 5]
```diff
@@ -130,6 +193,43 @@
...
+					lookbehind: true,
+					inside: {
+						'statement': {
+							pattern: /^(?:saveresult)(?:\s+[\S]+?)(?=\s)/i,
```
Comment (RunDevelopment) [inline components/prism-sas.js]: 1. `[\S]` == `\S`
2. `\S+?(?=\s)` == `\S+`

```suggestion
							pattern: /^saveresult\s+\S+/i,
```

File: components/prism-sas.js [hunk 6]
```diff
@@ -130,6 +193,43 @@
...
+							inside: {
+								keyword: /^(?:saveresult)/i
+							}
+						},
+						rest: args
+					}
+				},
+				'cas-actions': casActions,
+				'statement': {
+					pattern: /((?:^|\s)=?)(?:default|(?:un)?set|on|output|upload)(?:[\s\S]+?)(?=;)/im,
```
Comment (RunDevelopment) [inline components/prism-sas.js]:
```suggestion
					pattern: /((?:^|\s)=?)(?:default|(?:un)?set|on|output|upload)[^;]+/im,
```

File: components/prism-sas.js (hunk 1 above)
Comment (cedporter) [reply components/prism-sas.js]: I believe I intended to use it elsewhere eventually, but if/when that comes up, I can recreate this variable.

File: components/prism-sas.js (hunk 2 above)
Comment (cedporter) [reply components/prism-sas.js]: Incorporated in commit.

File: components/prism-sas.js (hunk 3 above)
Comment (cedporter) [reply components/prism-sas.js]: Incorporated in commit.

File: components/prism-sas.js (hunk 4 above)
Comment (cedporter) [reply components/prism-sas.js]: Incorporated in commit.

File: components/prism-sas.js (hunk 5 above)
Comment (cedporter) [reply components/prism-sas.js]: Incorporated in commit.

File: components/prism-sas.js (hunk 6 above)
Comment (cedporter) [reply components/prism-sas.js]: Incorporated in commit.

**Comment by cedporter** [comment]
It actually appears that the CI build has already passed, not sure why it's not reflected here. I think I've incorporated all requested changes. Thanks!

**Comment by RunDevelopment** [comment]
Thank you for contributing!

**Pull request merged**